include/zephyr/drivers/sensor_v2.h

```diff
@@ -0,0 +1,989 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file drivers/sensor_v2.h
+ *
+ * @brief Public APIs for the sensor driver.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENSOR_V2_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENSOR_V2_H_
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENSOR_H_
+#error "Please include drivers/sensor.h directly"
+#endif
+
+/**
+ * @brief Sensor Interface V2
+ * @defgroup sensor_interface Sensor Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor_types.h>
+#include <zephyr/drivers/sensor_utils.h>
+#include <zephyr/math/util.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * The units in which a sensor represents its range.
+ */
+enum sensor_range_units {
+	/** Gravity units, may be used in SENSOR_TYPE_ACCELEROMETER */
+	SENSOR_RANGE_UNITS_ACCEL_G,
+	/** \f$m/s^2\f$, may be used in SENSOR_TYPE_ACCELEROMETER */
+	SENSOR_RANGE_UNITS_ACCEL_MS2,
+	/** Radians, may be used in SENSOR_TYPE_GYROSCOPE */
+	SENSOR_RANGE_UNITS_ANGLE_RADIANS,
+	/** Rotational degrees, may be used in SENSOR_TYPE_GYROSCOPE */
+	SENSOR_RANGE_UNITS_ANGLE_DEGREES,
+	/** &micro;Tesla, may be used in SENSOR_TYPE_GEOMAGNETIC_FIELD */
+	SENSOR_RANGE_UNITS_GEOMAGNETIC_MICRO_TESLAS,
+	/** Pascals, may be used in SENSOR_TYPE_PRESSURE */
+	SENSOR_RANGE_UNITS_PRESSURE_PASCALS,
+	/** Lux, may be used in SENSOR_TYPE_LIGHT */
+	SENSOR_RANGE_UNITS_LIGHT_LUX,
+	/**
+	 * Celsius, may be used in:
+	 * - SENSOR_TYPE_ACCELEROMETER_TEMPERATURE
+	 * - SENSOR_TYPE_GYROSCOPE_TEMPERATURE
+	 * - SENSOR_TYPE_GEOMAGNETIC_FIELD_TEMPERATURE
+	 */
+	SENSOR_RANGE_UNITS_TEMPERATURE_C,
+	/**
+	 * Kelvin, may be used in:
+	 * - SENSOR_TYPE_ACCELEROMETER_TEMPERATURE
+	 * - SENSOR_TYPE_GYROSCOPE_TEMPERATURE
+	 * - SENSOR_TYPE_GEOMAGNETIC_FIELD_TEMPERATURE
+	 */
+	SENSOR_RANGE_UNITS_TEMPERATURE_K,
```

Comment:
how user can define it's own unit? Or will it again requite edit of header in OS to define an application specific unit?

Reply:
I'm open to moving this to `#define` like the types are. I assumed there are much more finite units so we can expect people to push them upstream. Let me know if you'd like me to make that change.

Reply:
I recommend to leave the end user with ability to create own units easy. Not everybody wants to be involved with upstream process and going this way will allow people to easily create own drivers in private forks or out-of-tree.